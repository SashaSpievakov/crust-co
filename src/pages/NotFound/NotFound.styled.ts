import styled from 'styled-components';

const Wrapper = styled.article`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin-bottom: 75px;

  @media screen and (max-width: 900px) {
    margin-bottom: 50px;
  }

  @media screen and (max-width: 650px) {
    margin-bottom: 20px;
  }
`;

const Title = styled.h1`
  font-size: 64px;
  text-align: center;
  padding: 150px 0;

  @media screen and (max-width: 900px) {
    font-size: 50px;
    padding: 100px 0;
  }

  @media screen and (max-width: 650px) {
    font-size: 35px;
    padding: 50px 0;
  }

  @media screen and (max-width: 500px) {
    padding: 40px 0;
  }
`;

export { Title, Wrapper };
