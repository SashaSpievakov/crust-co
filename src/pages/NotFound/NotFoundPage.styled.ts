import styled from 'styled-components';

const Article = styled.article`
  display: flex;
  flex-direction: column;
  align-items: center;
  margin: 120px 0 50px;

  p {
    margin-bottom: 50px;
  }

  @media screen and (max-width: 600px) {
    margin-top: 60px;
  }
`;

const Title = styled.h2`
  font-size: 60px;
  line-height: 1.1;
  text-align: center;
  margin-bottom: 20px;

  @media screen and (max-width: 750px) {
    font-size: 45px;
    padding: 0 15px;
  }

  @media screen and (max-width: 400px) {
    font-size: 35px;
  }
`;

export { Article, Title };
