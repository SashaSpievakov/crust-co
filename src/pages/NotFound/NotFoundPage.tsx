import { ErrorParagraph } from '@src/styles/Base.styled';
import { FC } from 'react';

import { ButtonLink } from '../../components/UI';
import { Article, Title } from './NotFoundPage.styled';

export const NotFoundPage: FC = () => {
  return (
    <Article>
      <Title>Oops! This page doesn&apos;t exist</Title>
      <ErrorParagraph>
        We couldn&apos;t find the page you were looking for. Please check the
        URL or navigate back to the homepage.
      </ErrorParagraph>
      <ButtonLink link="/">Return Home</ButtonLink>
    </Article>
  );
};
