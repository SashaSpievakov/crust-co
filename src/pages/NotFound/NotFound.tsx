import { ErrorParagraph } from 'src/styles/Base.styled';

import { ButtonLink } from '../../components/UI';
import { Article, Title } from './NotFound.styled';

const NotFound = () => {
  return (
    <Article>
      <Title>The page wasn&apos;t found</Title>
      <ErrorParagraph>Make sure you wrote the correct URL</ErrorParagraph>
      <ButtonLink link="/">Go Back</ButtonLink>
    </Article>
  );
};

export default NotFound;
