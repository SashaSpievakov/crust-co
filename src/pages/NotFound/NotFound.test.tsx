import '@testing-library/jest-dom';
import { screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import NotFound from './NotFound';
import rendererWithProviders from '../../tests/helpers/rendererWithProviders';
import renderWithProviders from '../../tests/helpers/renderWithProviders';

describe('NotFound Tests', () => {
  test('renders the NotFound page', () => {
    const snapshot = rendererWithProviders(<NotFound />);
    expect(snapshot).toMatchSnapshot();
  });

  test('checks redirect to the Home page', () => {
    renderWithProviders(null, true, '/notfound');
    const link = screen.getByRole('link', {
      name: /go back/i,
    });

    userEvent.click(link);

    expect(screen.getByText(/all pizzas/i)).toBeInTheDocument();
  });
});
