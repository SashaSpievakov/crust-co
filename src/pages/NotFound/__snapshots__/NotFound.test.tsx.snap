// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Tests renders the NotFound page 1`] = `
Object {
  "children": Array [
    <h2
      className="sc-ipEyDJ buSLWO"
    >
      The page wasn't found
    </h2>,
    <p
      className="sc-gswNZR cyBoAG"
    >
      Make sure you wrote the correct URL
    </p>,
    <a
      className="sc-hLBbgP sc-eDvSVe fGqcTx lhIsum"
      href="/"
      onClick={[Function]}
    >
      Go Back
    </a>,
  ],
  "props": Object {
    "className": "sc-kDvujY datOMI",
  },
  "type": "article",
}
`;
